src/main/kotlin/no/skatteetaten/aurora/boober/service/AuroraDeploymentSpecService.kt

```diff
@@ -0,0 +1,38 @@
+package no.skatteetaten.aurora.boober.service
+
+import no.skatteetaten.aurora.boober.mapper.v1.createAuroraDeploymentSpec
+import no.skatteetaten.aurora.boober.model.ApplicationId
+import no.skatteetaten.aurora.boober.model.AuroraConfig
+import no.skatteetaten.aurora.boober.model.AuroraDeploymentSpec
+import org.springframework.stereotype.Service
+
+@Service
+class AuroraDeploymentSpecService(val auroraConfigService: AuroraConfigService) {
+
+    fun getAuroraDeploymentSpecsForEnvironment(auroraConfigName: String, environment: String): List<AuroraDeploymentSpec> {
+        val auroraConfig = auroraConfigService.findAuroraConfig(auroraConfigName)
+        return auroraConfig.getApplicationIds()
+                .filter { it.environment == environment }
+                .let { getAuroraDeploymentSpecsForApplicationIds(auroraConfig, it) }
+    }
+
+    fun getAuroraDeploymentSpecs(auroraConfigName: String, aidStrings: List<String>): List<AuroraDeploymentSpec> {
+        val auroraConfig = auroraConfigService.findAuroraConfig(auroraConfigName)
+        return aidStrings.map(ApplicationId.Companion::fromString)
+                .let { getAuroraDeploymentSpecsForApplicationIds(auroraConfig, it) }
+    }
+
+    fun getAuroraDeploymentSpec(auroraConfigName: String, environment: String, application: String): AuroraDeploymentSpec {
+        val auroraConfig = auroraConfigService.findAuroraConfig(auroraConfigName)
+        return createAuroraDeploymentSpec(auroraConfig, ApplicationId.aid(environment, application))
+    }
+
+    private fun getAuroraDeploymentSpecsForApplicationIds(auroraConfig: AuroraConfig, applicationIds: List<ApplicationId>): List<AuroraDeploymentSpec> {
```

Comment:
Jeg tror dette er en god måte å gjøre det på. Jeg har egentlig ønsket å gjøre denne refaktoriseringen selv, spesielt etter @bjartek introduserte APPLICATION_PLATFORM_HANDLERS.

Det jeg tror kan være lurt å gjøre videre i denne refaktoriseringen er å flytte ansvaret for å kalle AuroraDeploymentSpecBuilder.createAuroraDeploymentSpec fra AuroraConfig til AuroraDeploymentSpecService. Jeg tror også vi bør flytte eierskap for APPLICATION_PLATFORM_HANDLERS inn hit og sørger for at initialiseringen skjer i forbindelse med life cycle til AuroraDeploymentSpecService. Slik koden er strukturert nå har vi egentlig global state som må være initialisert for at AuroraDeploymentSpecBuilder.createAuroraDeploymentSpec skal virke.

Det er mao noen implisitte, diffuse, men harde bindinger mellom helt separate deler av koden.

Reply:
@bentsolheim jeg har sjekket inn endringene vi diskuterte, kan du ta en titt på AuroraDeploymentSpecService igjen og se om dette er ok nå?

Som vi snakket om så er dette bare første steget, med å få flyttet APPLICATION_PLATFORM_HANDLERS inn i AuroraDeploymentSpecService. Så kan vi senere ta en vurdering på hvordan dette kan gjøres bedre senere.